docs/operate-and-deploy/migrations-tool.md

```diff
@@ -0,0 +1,609 @@
+project, and are named according to the convention `V<six digit version>__<description>.sql`. 
+An example directory structure is as follows:
+
+```
+<migrations-project-dir>
+|
+|- ksql-migrations.properties
+|- migrations/
+   |
+   |- V000001__Initial_setup.sql
+   |- V000002__Add_users.sql
+```
+
+You can use the `ksql-migrations create` command to easily create a blank migration
+file according to the naming scheme above, which can then be populated with ksqlDB
+statements and applied to your ksqlDB cluster.
+
+```
+ksql-migrations {-c | --config-file} <config-file> create
+                [ {-v | --version} <version> ] [--] <description>
+```
+
+To use the `ksql-migrations create` command, pass in the path to the config file 
+of your migrations project along with a description for your new migration file. 
+You can optionally pass in a specific version number for the new file as well.
+If unspecified, the next available version number will be used. 
+Note that 0 is not a valid migration version.
+
+```
+$ ksql-migrations --config-file /my/migrations/project/ksql-migrations.properties create Add_users
+
+Creating migration file: /my/migrations/project/migrations/V000002__Add_users.sql
+Migration file successfully created
+Execution time: 0.0480 seconds
+```
+
+You can now populate the empty migrations file with ksqlDB statements and apply
+the migration to your cluster.
+
+Apply Migrations
+----------------
+
+The `ksql-migrations apply` command reads ksqlDB statements from your migration
+files and applies them to your ksqlDB cluster.
+
+```
+ksql-migrations {-c | --config-file} <config-file> apply
+                [ {-a | --all} ] 
+                [ {-n | --next} ] 
+                [ {-u | --until} <untilVersion> ] 
+                [ {-v | --version} <version> ]
+                [ --dry-run ] 
+```
+
+There are four different modes for specifying which migration file version(s)
+to apply:
+* `all`: Apply all available migration files, from the latest applied version.
+* `next`: Apply the next available migration file, from the latest applied version.
+* `until`: Apply all available migration files, from the latest applied version
+  through the specified `untilVersion`.
+* `version`: Apply the migration file with the specified version only. The supplied
+  version cannot be older than the latest applied version.
+  
+In addition to selecting a mode for `ksql-migrations apply`, you must also pass 
+in the path to the config file of your migrations project as part of the command.
+
+You can optionally use the `--dry-run` flag to see which migration file(s) the
+command will apply before running the actual `ksql-migrations apply` command
+to update your ksqlDB cluster. Note that the dry run does not validate whether
+the ksqlDB server will accept the statements in your migration file(s); the dry
+run only serves to display which commands the migrations tool will attempt to
```

Comment:
I left the word "only" in (despite accepting the rest of the suggestion) since I think it's important to clarify that the dry-run is very limited in what it actually does. I think this will be unintuitive to many users.